cpp/include/raft/sparse/linalg/spmm.hpp

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, NVIDIA CORPORATION.
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
```

Comment:
Since this is a publicly facing header, we should add a new file called `spmm.cuh` and have it import this header (with an include guard, of course), so that it doesn't break users downstream.

Reply:
Make sense.